internal/services/containers/kubernetes_flux_configuration_resource.go

```diff
@@ -244,7 +243,7 @@ func (r KubernetesFluxConfigurationResource) Arguments() map[string]*pluginsdk.S
 					"container_id": {
 						Type:         pluginsdk.TypeString,
 						Required:     true,
-						ValidateFunc: storageValidate.StorageContainerDataPlaneID,
+						ValidateFunc: validation.IsURLWithPath, // note: storage domain suffix cannot be determined at validation time, so just make sure it's a well-formed URL
```

Comment:
Whilst we don't have access to the Environment that doesn't mean we couldn't necessarily do an opt-in enhanced validation here? e.g. expose a global variable for `StorageDomainSuffix` that's set in a similar manner as we do for `Location`s and then conditionally validate the domain name, else just check the Path matches?

Reply:
I couldn't see a race-free way to achieve this, since the `environment` (cloud) can be set in provider configuration as well as via environment variable - building the provider also occurs post-validation so we still can't work out what the appropriate domain should be?

Reply:
Seems to work with populating a global variable! Validation function is updated accordingly :)